src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -160,46 +161,69 @@ public static object ConvertFromJson(string input, bool returnHashtable, int? ma
             error = null;
             try
             {
-                // JsonConvert.DeserializeObject does not throw an exception when an invalid Json array is passed.
-                // This issue is being tracked by https://github.com/JamesNK/Newtonsoft.Json/issues/1930.
-                // To work around this, we need to identify when input is a Json array, and then try to parse it via JArray.Parse().
 
-                // If input starts with '[' (ignoring white spaces).
-                if (Regex.Match(input, @"^\s*\[").Success)
-                {
-                    // JArray.Parse() will throw a JsonException if the array is invalid.
-                    // This will be caught by the catch block below, and then throw an
-                    // ArgumentException - this is done to have same behavior as the JavaScriptSerializer.
-                    JArray.Parse(input);
-
-                    // Please note that if the Json array is valid, we don't do anything,
-                    // we just continue the deserialization.
-                }
+                var serializer = new JsonSerializer();
 
-                var obj = JsonConvert.DeserializeObject(
-                    input,
-                    new JsonSerializerSettings
+                serializer.TypeNameHandling = TypeNameHandling.None;
+                serializer.MetadataPropertyHandling = MetadataPropertyHandling.Ignore;
+                serializer.MaxDepth = maxDepth;
+                using (System.IO.MemoryStream stream = new System.IO.MemoryStream(System.Text.Encoding.Default.GetBytes(input)))
+                {
+                    using (System.IO.StreamReader streamReader = new System.IO.StreamReader(stream))
                     {
-                        // This TypeNameHandling setting is required to be secure.
-                        TypeNameHandling = TypeNameHandling.None,
-                        MetadataPropertyHandling = MetadataPropertyHandling.Ignore,
-                        MaxDepth = maxDepth
-                    });
+                        using (JsonReader reader = new JsonTextReader(streamReader))
```

Comment:
I'm not sure which version of C# this project uses, but I _think_ it's C# 9, in which case you can use
```
using System.IO.MemoryStream stream = new System.IO.MemoryStream(System.Text.Encoding.Default.GetBytes(input));
using System.IO.StreamReader streamReader = new System.IO.StreamReader(stream);
using JsonReader reader = new JsonTextReader(streamReader);
```
instead to reduce a lot of the nesting.

Reply:
done!